src/main/java/com/amazonaws/services/dynamodbv2/AmazonDynamoDBLockClient.java

```diff
@@ -470,11 +471,27 @@ public LockItem acquireLock(final AcquireLockOptions options) throws LockNotGran
                     }
 
                     if (options.shouldSkipBlockingWait() && existingLock.isPresent() && !existingLock.get().isExpired()) {
+
+                        String id = existingLock.get().getUniqueIdentifier();
+                        // Let's check to see if this existingLock expired based on old data we cached.
+                        // Or cache it if we haven't seen this recordVersion before.
+                        boolean itReallyIsExpired = false;
+                        if (notMyLocks.containsKey(id) &&
+                              notMyLocks.get(id).getRecordVersionNumber()
+                              .equals(existingLock.get().getRecordVersionNumber())) {
+                          itReallyIsExpired = oldCopy.isExpired();
```

Comment:
I think this approach works, with the following downsides:
1. The very first acquire call on an existing lock will throw, always. There should be a separate check for `notMyLocks.containsKey(id)`.
2. Acquiring expired locks will take twice as long since `oldCopy.isExpired()` will return true only after `lease_duration` secs. Then the rest of the acquire will take another `lease_duration` seconds.

Reply:
There is another option which is to block on the first call. If it returned false, then the following calls should fail fast.

Reply:
If the user wants to block to wait for a lease, they should not set: ShouldSkipBlockingWait

Reply:
I am fine with that approach as long as it is made clear (in the README?) that a return value of false does not only mean the lease is held. It may also mean that the client is, with the information that it has, not able to determine whether the lease is owned or not.

Reply:
Probably not related to this PR, but instead of using the `skipBlockingWait` property on the client level there can be two acquire methods - one for blocking and another for non-blocking.

Reply:
See https://github.com/awslabs/amazon-dynamodb-lock-client/pull/102 for updates to the README.